homeassistant/components/mqtt/device_tracker.py

```diff
@@ -5,30 +5,58 @@
 
 from homeassistant.components import mqtt
 from homeassistant.components.device_tracker import PLATFORM_SCHEMA
+from homeassistant.components.device_tracker.const import (
+    SOURCE_TYPE_GPS,
+    SOURCE_TYPE_ROUTER,
+    SOURCE_TYPE_BLUETOOTH,
+    SOURCE_TYPE_BLUETOOTH_LE,
+)
 from homeassistant.const import CONF_DEVICES
 from homeassistant.core import callback
 import homeassistant.helpers.config_validation as cv
 
-from . import CONF_QOS
+from . import CONF_QOS, CONF_SOURCE_TYPE
 
 _LOGGER = logging.getLogger(__name__)
 
+_VALID_SOURCE_TYPE_SCHEMA = vol.All(
+    vol.Coerce(str),
```

Comment:
Just use `vol.In`, coercing will never lead to a valid type anyway.

Reply:
This should be addressed now.
I have added two tests as well.